### ISSUE TITLE
Any plan about the IDE's syntax highlighting support ?

### ISSUE THREAD

**Issue by ablipan** [mbosecke/pebble#173]
Pebble is robust, but write the template code without syntax highlighting is a little painful.
Is there any plan to add the IDE's syntax highlighting?
By the way, I'm using the IntelliJ IDEA.

**Comment by decebals** [comment]
+1 
It's an old idea of mine. Will be a fantastic tool. I can help here but I know nothing about the plugins system from IDEA. I can start looking at an open source plugin built for another template engine (freeamrker, velocity, jade, ...) and I will try to create something minimalist in first step. My problem is that I don't have time.

**Comment by mbosecke** [comment]
Yup, a syntax highlighting tool is something I've wanted since the beginning. Unfortunately, I also have no knowledge about creating IDE plugins and time is also not on my side. The plan so far has been to cross my fingers and hope that someone out there will be crazy enough to take on that responsibility.

**Comment by ablipan** [comment]
:smile: I've installed the twig plugin for IntelliJ just now, and works fine with pebble template.

**Comment by decebals** [comment]
I can confirm that you have some syntax highlights with twig plugin for IDEA.
My intention was to build something more powerful and customize for Pebble but I think that using twig plugin is a good compromise.

All you need to do is to add the custom extension (`.peb`) so IDEA can recognize template files.

Steps:

1. Install twig plugin
2. Go to `File -> Settings -> Editor -> File Types`
3. Find Twig, it's identical template engine but for PHP and add custom extension. click green plus and type `*.peb`. Now all `*.peb` files are parsed as Twig so in the end - support Pebble.

**Comment by KazeFlame** [comment]
+1 for Atom

**Comment by pun-ky** [comment]
I was using Twig Plugin for IntelliJ and it highlights perfectly but for most recent version of IDE it is not available :/

**Comment by mohamedb** [comment]
I just changed the extension from html to html.twig and installed Twig extension (Eclipse & Netbeans) and work perfectly 👍

**Comment by jochen777** [comment]
We changed the suffix to .twig as well and installed the plugin http://twig.dubture.com/installation/ for eclipse. Works okay.

**Comment by liias** [comment]
If anyone in interested in IntelliJ IDEA support specifically, please make a separate issue.
If there will be high amount of votes I might look into it.

**Comment by bjansen** [comment]
Hi guys!

I wanted to let you know that I published a new IntelliJ plugin that supports Pebble: https://plugins.jetbrains.com/idea/plugin/9407-pebble

Feedback is highly welcome, v0.2 is on the way (https://github.com/bjansen/pebble-intellij/milestone/2).

**Comment by ebussieres** [comment]
Awesome 👍

**Comment by decebals** [comment]
@bjansen I gave you a big star on https://github.com/bjansen/pebble-intellij. Thanks!
